airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -128,3 +130,66 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
                 ]
             )
         return streams
+
+    def _read_incremental(
```

Comment:
A few comments about this override:
1. please, everytime we override something fill the docstring explaining why the override is needed (especially private method)
2. I believe we can and should do error handling on the stream level (or `GoogleAds` class), not here
3. The pattern looks like typical retry case. I don't think reducing page size is a good solution because we basically lose all progress. I propose to retry query with new state and continue reading. The data in response returned in ascending order, therefore should be no problem to continue reading where we crashed.
WDYT?

Reply:
here is an explanation of 3rd point.
@augan-rymkhan 
Lets say we request the data:

from `2010-01-01 00:00:00`  to `2010-01-15 00:00:00`
we read 10 pages, but there is another 20 pages left
we crashed at `cursor_value` = `2010-01-05 13:00:12`
we will retry query
from `2010-01-05 13:00:12`  to `2010-01-15 00:00:00`.

if this will not work (I don't see why, but...), we can retry exactly the same query but continue from the pages number where we crashed.

Reply:
3. `stream_instance.stream_slices` receives `stream_state`, which is the latest state taken from [get_updated_state](https://github.com/airbytehq/airbyte/pull/9812/files#diff-4c20863cc2814c2d4d368f1106fd91b9fa1be29030e273caca0960de9967ccb2R167) and generate slices from that [point](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py#L83). It doesn't lose the progress, it regenerates slices with reduces date range twice starting from the latest read record's cursor value.